sql: TestRelocateNonVoters failed

sql.TestRelocateNonVoters [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21190691?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21190691?buildTab=artifacts#/) on master @ [ce737bca704957fcf69308781ce731ea7cdcc32c](https://github.com/cockroachdb/cockroach/commits/ce737bca704957fcf69308781ce731ea7cdcc32c):


```
=== RUN   TestRelocateNonVoters
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/230b14ff9b70e2729ccc9b9deabcc166/logTestRelocateNonVoters3889208977
    test_log_scope.go:82: use -show-logs to present logs inline
    multitenant_admin_function_test.go:987: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/230b14ff9b70e2729ccc9b9deabcc166/logTestRelocateNonVoters3889208977
--- FAIL: TestRelocateNonVoters (72.07s)
=== RUN   TestRelocateNonVoters/ALTER_RANGE_x_RELOCATE_NONVOTERS
    test_server_shim.go:686: DRPC is enabled (via override by TestingGlobalDRPCOption)
    multitenant_admin_function_test.go:937: condition failed to evaluate within 45s: from multitenant_admin_function_test.go:953: waiting for LEARNER replicas: {2}
        
        goroutine dump: /artifacts/tmp/_tmp/230b14ff9b70e2729ccc9b9deabcc166/goroutine_dump_826182237.txt
--- FAIL: TestRelocateNonVoters/ALTER_RANGE_x_RELOCATE_NONVOTERS (58.42s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRelocateNonVoters.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-61019

--- a/pkg/sql/multitenant_admin_function_test.go
+++ b/pkg/sql/multitenant_admin_function_test.go
@@ -347,15 +347,24 @@ func (tc testCase) runTest(
 		fn()
 	}
 
-	execQueries(testCluster, systemDB, "system", tc.system)
+	// Re-enable queues before each execQueries call so that queues
+	// disabled by a prior tenant's run don't block replication.
+	runExecQueries := func(db *gosql.DB, tenant string, tExp tenantExpected) {
+		testCluster.ToggleReplicateQueues(true)
+		testCluster.ToggleSplitQueues(true)
+		testCluster.ToggleLeaseQueues(true)
+		execQueries(testCluster, db, tenant, tExp)
+	}
+
+	runExecQueries(systemDB, "system", tc.system)
 	if tc.secondary.isSet() {
-		execQueries(testCluster, secondaryDB, "secondary", tc.secondary)
+		runExecQueries(secondaryDB, "secondary", tc.secondary)
 	}
 	if tc.secondaryWithoutClusterSetting.isSet() {
-		execQueries(testCluster, secondaryWithoutClusterSettingDB, "secondary_without_cluster_setting", tc.secondaryWithoutClusterSetting)
+		runExecQueries(secondaryWithoutClusterSettingDB, "secondary_without_cluster_setting", tc.secondaryWithoutClusterSetting)
 	}
 	if tc.secondaryWithoutCapability.isSet() {
-		execQueries(testCluster, secondaryWithoutCapabilityDB, "secondary_without_capability", tc.secondaryWithoutCapability)
+		runExecQueries(secondaryWithoutCapabilityDB, "secondary_without_capability", tc.secondaryWithoutCapability)
 	}
 }
 

